File: Homework/Mary_Antonyan/C++/27_03_2017/Overlapping/point.cpp
#include "point.hpp"

//Constructor
Point::Point(double x, double y) 
    : x(x)
    , y(y)
{}

//Get-Methods
double Point::getX() const {
    return x;
}

double Point::getY() const {
    return y;
}

//Destructor
Point::~Point() {}

File: Homework/Smbat/java/paint/paint/shapes/Point.java
package paint.shapes;

import java.io.PrintStream;
import java.util.Arrays;
import paint.Shape;

public class Point implements Shape {
    private int x;
    private int y;
    
    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }
    
    

    public boolean draw(PrintStream out, char symbol) {
        String[][] canvas = new String[100][100];
        for (String[] canva : canvas) {
            Arrays.fill(canva, "~");
        }
        canvas[y][x] = symbol + "";
        
        for (int i = 0; i < canvas.length; i++) {
            for (String pixel : canvas[i]) {
                out.print(pixel);
            }
            out.println();
        }
        return true;
    }
}